velox/exec/Operator.h

```diff
@@ -476,6 +478,13 @@ class Operator : public BaseRuntimeStatWriter {
   // 'identityProjections_' and 'resultProjections_'.
   RowVectorPtr fillOutput(vector_size_t size, BufferPtr mapping);
 
+  // Returns the max number of rows in the output batch.
+  virtual uint32_t outputBatchRows() const;
+
+  // Estimated size of an output row. Returns the default value -1 if this
```

Comment:
Would it make sense to return an std::optional<int64_t> instead of using a special value for unknown?